common/remote/sockfile.cpp

```diff
@@ -2927,11 +2927,12 @@ inline void appendErr3(MemoryBuffer &reply, RemoteFileCommandType e, int code, c
 
 
 
-#define MAPCOMMAND(c,p) case c: { ret =  this->p(msg, reply) ; break; }
-#define MAPCOMMANDCLIENT(c,p,client) case c: { ret = this->p(msg, reply, client); break; }
-#define MAPCOMMANDCLIENTTHROTTLE(c,p,client,throttler) case c: { ret = this->p(msg, reply, client, throttler); break; }
-#define MAPCOMMANDSTATS(c,p,stats) case c: { ret = this->p(msg, reply, stats); break; }
-#define MAPCOMMANDCLIENTSTATS(c,p,client,stats) case c: { ret = this->p(msg, reply, client, stats); break; }
+#define DEFINESTATS(c) ClientStats *stats = clientStatsTable.addClientReference(c, client->queryPeerName())
+#define MAPCOMMAND(c,p) case c: { DEFINESTATS(c); ret = this->p(msg, reply) ; break; }
+#define MAPCOMMANDCLIENT(c,p,client) case c: { DEFINESTATS(c); ret = this->p(msg, reply, client); break; }
+#define MAPCOMMANDCLIENTTHROTTLE(c,p,client,throttler) case c: { DEFINESTATS(c); ret = this->p(msg, reply, client, throttler); break; }
```

Comment:
All commands need it, some augment the stats in the cmd (MAPCOMMANDSTATS/MAPCOMMANDCLIENTSTATS varieties do), but all want to add a reference in the clientStatsTable.

Previously it was outside and before the case statement, which meant garbage commands with cmd indexes out of range, were being added, with the result a cmd array of counts would be access out of bounds in addClientCommon()
This line:
```
        cmdStats[cmd]++;
```

So now only the clientStatsTable via these macros has moved inside the case blocks and won't happen on an invalid command.